python-cell.el

```diff
@@ -51,10 +51,14 @@
   :safe 'booleanp)
 
 (defcustom python-cell-cellbreak-regexp
-  (rx line-start (* space)
-      (group (and "#" (or (and "#" space (* (not (any "\n"))))
-                          (and " <" (or "codecell" "markdowncell") ">"))
-                  line-end)))
+  (rx line-start 
+      (or (and (* space)
+               (group-n 1 (and "#" (or (and "#" space (* (not (any "\n"))))
+                                       (and " <" (or "codecell" "markdowncell") ">"))
+                               line-end)))
+          (group-n 1 (and "if False:"
```

Comment:
We could allow any number of spaces between `if` and `False` but that would make the logic inside `python-cell-preprocess-cell` more complicated. And I don't know if there would be a need for any other code formatting than `if False:` so maybe it is good enough to only detect exactly this string.

Reply:
I think it suffices to hard code "if False:" as you did. Why do you have to use `group-n` in the `rx` macro, isn't the following sufficient?

```lisp
  (rx line-start (* space)
      (group (and (or (and "#" (or (and "#" space (* (not (any "\n"))))
                                   (and " <" (or "codecell" "markdowncell") ">")))
                      "if False:")
                  line-end)))
```

Reply:
That is a good point. I initially wrote something similar to that code. But I think that this version with only one group is going to detect also rich comments preceded by space, e.g. `       if False:`. I wanted to prohibit that, because evaluating that cell would probably be invalid syntax unless we remove the indentation before sending it to the Python interpreter. If we were to support indentation in front of `if False:`, we could maybe have cells defined *inside* other code such as functions, but I think that would complicate things and not be very useful, so I disallow it.

The `(group-n 1 ...)` is needed over `(group ...)` in order for the code that I wrote to work properly.